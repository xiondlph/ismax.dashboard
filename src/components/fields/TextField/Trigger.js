/**
 * Стандартный триггер
 */

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

class Trigger extends Component {
    onIconClick() {
        const { handler } = this.props

        if (typeof handler === 'function') {
            handler()
        }
    }

    render() {
        const { tip, icon, type } = this.props

        return (
            <div className={classNames({
                'dashboard-field-text-trigger': true,
                [`dashboard-field-text-trigger_${type}`]: !!type
            })}>
                <span
                    className={`dashboard-field-text-trigger-icon fa fa-${icon}`}
                    data-tip={tip}
                    data-type={type}
                    onClick={::this.onIconClick}
                />
            </div>
        )
    }
}

Trigger.propTypes = {
    tip: PropTypes.string.isRequired,
    icon: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    handler: PropTypes.func
}

export default Trigger