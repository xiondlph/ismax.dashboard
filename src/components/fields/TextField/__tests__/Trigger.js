import React from 'react'
import renderer from 'react-test-renderer'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Trigger from '../Trigger'

configure({ adapter: new Adapter() })

describe('Trigger', () => {
    test('Проверка Trigger снапшота', () => {
        const
            tip = 'trigger',
            icon = 'icon',
            type = 'default',
            handler = jest.fn(),
            component = renderer.create(
                <Trigger tip={tip} icon={icon} type={type} handler={handler} />
            )

        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })

    test('Проверка колбека на клик', () => {
        const
            tip = 'trigger',
            icon = 'icon',
            type = 'default',
            handler = jest.fn(),
            trigger = shallow(
                <Trigger tip={tip} icon={icon} type={type} handler={handler} />
            );

        trigger.find('.dashboard-field-text-trigger-icon').simulate('click')
        expect(handler.mock.calls.length).toBe(1)
    })
})