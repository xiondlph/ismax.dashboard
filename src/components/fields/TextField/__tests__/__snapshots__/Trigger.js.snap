// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trigger Проверка Trigger снапшота 1`] = `
<div
  className="dashboard-field-text-trigger dashboard-field-text-trigger_default"
>
  <span
    className="dashboard-field-text-trigger-icon fa fa-icon"
    data-tip="trigger"
    data-type="default"
    onClick={[Function]}
  />
</div>
`;
